crates/core/types/account.rs

```diff
@@ -108,26 +108,28 @@ impl RLPDecode for AccountState {
     }
 }
 
-pub fn compute_storage_root(storage: &HashMap<H256, H256>) -> H256 {
-    let rlp_storage = storage
-        .iter()
-        .map(|(k, v)| {
-            let mut k_buf = vec![];
-            let mut v_buf = vec![];
-            k.encode(&mut k_buf);
-            v.encode(&mut v_buf);
-            (k_buf, v_buf)
-        })
-        .collect::<Vec<_>>();
-    let root =
-        PatriciaMerkleTree::<_, _, Keccak256>::compute_hash_from_sorted_iter(rlp_storage.iter());
+pub fn compute_storage_root(storage: &BTreeMap<H256, H256>) -> H256 {
+    let mut pmt = PatriciaMerkleTree::<Vec<u8>, Vec<u8>, Keccak256>::new();
```

Comment:
nit: the short way to describe a patricia merkle tree is `trie`, or in this case maybe `storage_trie`